core/common/src/main/java/alluxio/exception/AlluxioRuntimeException.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.exception;
+
+import com.google.common.base.Preconditions;
+import io.grpc.Status;
+import io.grpc.StatusException;
+
+/**
+ * Alluxio RuntimeException. Every developer should throw this exception when need to surface
+ * exception to client.
+ */
+public class AlluxioRuntimeException extends RuntimeException {
+  private static final long serialVersionUID = 7801880681732804395L;
+  private final Status mStatus;
```

Comment:
This is a good start, but we probably want to also add more details similar to https://cloud.google.com/apis/design/errors#error_model. Since details can be an arbitrary protobuf, we can just use an Object on the base class (or is there a common abstract class for all protobuf?). Each child class can populate the right details according to the above mentioned doc.